src/analysis/event_detector.py

```diff
@@ -0,0 +1,312 @@
+import xarray as xr
+import numpy as np
+from pathlib import Path
+from xclim.run_length import rle, longest_run  # , windowed_run_events
+from typing import Tuple, Optional, Any
+import warnings
+import pandas as pd
+from scripts.eng_utils import get_ds_mask
+
+
+class EventDetector():
```

Comment:
Can something be added to explain what happens here?

Also to specify that this only works with CHIRPS preprocessed data

Reply:
it works also for VHI? or any other netcdf file

Reply:
added docstring